### PR TITLE
Add SIMD-0007: Access Policy

### PR DESCRIPTION
**Pull request by jacobcreech**
#### Problem

There isn't a clear answer today on how to grant triage, write, or maintain access to review SIMD pull requests.

#### Proposed Solution

I have drafted up an access policy, shamelessly borrowing from the [Mozilla Access Policy](https://www.mozilla.org/en-US/about/governance/policies/commit/access-policy/)

### CODE REVIEW

File: proposals/0007-access-policy.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+### Level 3 - Maintain Access
+
+Requirements: One voucher from anyone with level 3 access.
+
+Contributors will have the associated [maintain Github access
+policy](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role)
+for the [Solana Improvement
+Documents](https://github.com/solana-foundation/solana-improvement-documents)
+repository. They will also have the Contributor role in the Solana Tech
```
Comment (jacobcreech) [inline proposals/0007-access-policy.md:112]: Don't think role in discord matters as much for those with maintain access.

File: proposals/0007-access-policy.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+2. Core
+
+    Requirement: One voucher from anyone with level 3 access or two from
+    anyone with level 2 access.
+
+3. Maintain
+
+    Requirement: One voucher from anyone with level 3 access. This permission
+    is usually reserved for those maintaining the SIMD process.
```
Comment (mvines) [inline proposals/0007-access-policy.md:28]: What's the process for removing problematic users? I feel that one user with Maintain access could remove a Core or Triage user. Maintainers that want to remove other maintainers must cage match

File: proposals/0007-access-policy.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+increase in governance overhead.
+
+## New Terminology
+
+Core Contributor - Individuals with Core access to SIMDs that can assist in
+triaging different SIMDs.
+
+## Detailed Design
+
+### Level 1 - Triage Access
```
Comment (mvines) [inline proposals/0007-access-policy.md:78]: How were you thinking of actually implementing the vouching process? Again stealing from Mozilla, they do this by way of bug tracker:
1. The person requesting access opens a new issue, with their specific request
2. Voucher does so by commenting on the issue
3. Devops then sees the vouch and implements the access change

Removes are documented in a similar way. 

This generates a really nice audit log for access changes

File: proposals/0007-access-policy.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+3 levels of access, in the order of increasing access:
+
+1. Triage
+
+    Requirement: One voucher from anyone with level 2 or above access.
+
+2. Core
+
+    Requirement: One voucher from anyone with level 3 access or two from
```
Comment (mvines) [inline proposals/0007-access-policy.md:22]: I'd suggest that for Core access, you also first need Triage access. So Level 2 is like a promotion from level 1, you can't just start at Level 2

File: proposals/0007-access-policy.md (hunk 1 above)
Comment (mvines) [reply proposals/0007-access-policy.md:112]: I feel the Solana Tech Discord roles are an independent matter entirely, which also needs a formal ACL process of course.

File: proposals/0007-access-policy.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+## Summary
+
+3 levels of access, in the order of increasing access:
+
+1. Triage
+
+    Requirement: One voucher from anyone with level 2 or above access.
+
+2. Core
```
Comment (mvines) [inline proposals/0007-access-policy.md:20]: Consider renaming "Core" to "Write"? Now we essentially have something that directly maps into the standard github repository roles

File: proposals/0007-access-policy.md (hunk 3 above)
Comment (jacobcreech) [reply proposals/0007-access-policy.md:78]: I was going to do exactly this. Build a git bot that watches issues for vouchers and maintains the addition/removal of access.

Comment (mvines) [reply proposals/0007-access-policy.md:78]: Using humans seems fair enough to start with too, building a bot seems great until it becomes a bigger burden than a meat-powered process. Just my immediate thought on that

Comment (mvines) [reply proposals/0007-access-policy.md:78]: That seems fine for now tbh, a small list :)
Your choice of course, just sharing what I'd do to start with. I find it hard to imagine in the short term there'll be an unmanageable flow of requests for access. I'd like to be proven wrong though!

File: proposals/0007-access-policy.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+1. Triage
+
+    Requirement: One voucher from anyone with level 2 or above access.
+
+2. Write
+
+    Requirement: One voucher from anyone with level 3 access or two from
+anyone with level 2 access. If only vouched by those with level 2 access, the
+user must have triage access to be promoted.
```
Comment (mvines) [inline proposals/0007-access-policy.md:24]:
```suggestion
user must have Triage access to be promoted.
```

File: proposals/0007-access-policy.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+granular labeling of different SIMDs.
+
+[PEP](https://peps.python.org/pep-0013/) has an election of a council that
+reviews and triages new proposed changes. The PEP approach may result in an
+increase in governance overhead.
+
+## New Terminology
+
+Core Contributor - Individuals with Write access to SIMDs that can assist in
+triaging different SIMDs.
```
Comment (mvines) [inline proposals/0007-access-policy.md:75]: Fortunately we managed to kill this new terminology

File: proposals/0007-access-policy.md (hunk 2 above)
Comment (jacobcreech) [reply proposals/0007-access-policy.md]: Added some clarification around revoking access. LMK what you think. No cage matches unfortunately.

File: proposals/0007-access-policy.md (hunk 3 above)
Comment (jacobcreech) [reply proposals/0007-access-policy.md:79]: Updated with Access request process, also stolen from Mozilla.

File: proposals/0007-access-policy.md (hunk 4 above)
Comment (jacobcreech) [reply proposals/0007-access-policy.md:22]: Added promotion requirement.

File: proposals/0007-access-policy.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+### Vouching Process 
+
+To receive a level of access as defined in SIMD-0007, follow these steps:
+
+1. Open an issue on the [Solana Improvement
+Documents](https://github.com/solana-foundation/solana-improvement-documents)
+repository with a title in the following format: "Request Access (Level [X])
+for [Username]".
+2. Gather your vouchers to add a comment on the issue expressing their support.
+3. Once the issue has received enough support, you will be granted the
```
Comment (mvines) [inline proposals/0007-access-policy.md:128]: How about pinging a "@SIMD-Maintainers" or something github team at this step too, to try to help summon a Maintainer.

File: proposals/0007-access-policy.md (hunk 5 above)
Comment (jacobcreech) [reply proposals/0007-access-policy.md]: Renamed. Originally wanted to make it obvious that "Core Contributors" had "Core Access", but the longer I thought about it the name only caused more confusion.

File: proposals/0007-access-policy.md (hunk 1 above)
Comment (jacobcreech) [reply proposals/0007-access-policy.md:108]: Should the Solana Tech Discord only be concerned with the Solana Labs validator implementation?

Comment (mvines) [reply proposals/0007-access-policy.md:108]: I do think there's a discussion to be had about what Solana Tech Discord and its roles ought to morph into as we clarify the lines between the various entities. New channels could perhaps be created for discussing proposals, for folks that prefer to discuss some points outside of github issues.

The "Core Contributor role" in the Solana Tech Discord essentially means write access to the solana-labs/solana monorepo and the SPL repo right now. Until Solana Tech Discord is cleaned up, I think granting that role to people who achieve write access here will be making the problem worse.

File: proposals/0007-access-policy.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+Discord.
+
+Users with Maintain access are responsible for managing the SIMD repository and
+granting user access based on the above requirements.
+
+### Vouching Process 
+
+To receive a level of access as defined in SIMD-0007, follow these steps:
+
+1. Open an issue on the [Solana Improvement
```
Comment (mvines) [inline proposals/0007-access-policy.md:118]: In the interest of potentially assigning a Discord role in the future, the requester could also provide their discord id in the issue

**Review by mvines — APPROVED** [review]
Looks great. 

To help implement this, you might consider adding [issue templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository) to help noobs format their access requests appropriately - a template can also do things like add a label and assign a user (the maintainers) to the issue.

**Review by mvines — APPROVED** [review]

**Comment by mvines** [comment]
welp I guess I get to use this policy once it's implemented!
image

**Comment by jacobcreech** [comment]
Merging this meta proposal based on feedback. We can update based on how well it works later.

File: proposals/0007-access-policy.md [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+    Requirement: One voucher from anyone with level 3 access. This permission
+is usually reserved for those maintaining the SIMD process.
+
+Each of the levels implies having the previous levels - e.g. level 2 implies
+level 1. A contributor with Write access will be able to triage pull requests
+in accordance with Github access policies.
+
+## Motivation
+
+Shamelessly borrowing from [Mozilla's access
```
Comment (hydrogenbond007) [inline proposals/0007-access-policy.md:37]: love the honesty 😂

Comment (jacobcreech) [reply proposals/0007-access-policy.md:37]: @hydrogenbond007 please don't make comments that do not contribute to the discussion on SIMDs. It is not the place for them.

**Pull request merged**